libs/framework/src/module.c

```diff
@@ -319,4 +351,172 @@ array_list_pt module_getDependents(module_pt module) {
+        *handle = libHandle;
+    }
+    return status;
+}
+
+
+static celix_status_t celix_module_loadLibraryForManifestEntry(celix_module_t* module, const char *library, bundle_archive_pt archive, void **handle) {
+    celix_status_t status = CELIX_SUCCESS;
+
+    const char *error = NULL;
+    char libraryPath[512];
+    const char* revRoot = celix_bundleArchive_getCurrentRevisionRoot(archive);
+
+    if (!revRoot) {
+        fw_logCode(module->fw->logger, CELIX_LOG_LEVEL_ERROR, status, "Could not get bundle archive information");
+        return status;
+    }
+
+    char* path;
+    if (strstr(library, CELIX_LIBRARY_EXTENSION)) {
+        path = celix_utils_writeOrCreateString(libraryPath, sizeof(libraryPath), "%s/%s", revRoot, library);
+    } else {
+        path = celix_utils_writeOrCreateString(libraryPath, sizeof(libraryPath), "%s/%s%s%s", revRoot, CELIX_LIBRARY_PREFIX, library, CELIX_LIBRARY_EXTENSION);
+    }
+
+    if (!path) {
+        fw_logCode(module->fw->logger, CELIX_LOG_LEVEL_ERROR, status, "Cannot create full library path");
+        return status;
+    }
+
+    status = celix_module_loadLibrary(module, path, handle);
+    celix_utils_freeStringIfNeeded(libraryPath, path);
+    framework_logIfError(module->fw->logger, status, error, "Could not load library: %s", libraryPath);
+    return status;
+}
+
+static celix_status_t celix_module_loadLibrariesInManifestEntry(celix_module_t* module, const char *librariesIn, const char *activator, bundle_archive_pt archive, void **activatorHandle) {
+    celix_status_t status = CELIX_SUCCESS;
+    celix_bundle_context_t* fwCtx = celix_framework_getFrameworkContext(module->fw);
+
+    char* last;
+    char* libraries = strndup(librariesIn, 1024*10);
+    char* token = strtok_r(libraries, ",", &last);
+    while (token != NULL) {
+        void *handle = NULL;
+        char lib[128];
+        lib[127] = '\0';
+
+        char *path = NULL;
+        char *pathToken = strtok_r(token, ";", &path);
+        strncpy(lib, pathToken, 127);
+        pathToken = strtok_r(NULL, ";", &path);
+
+        while (pathToken != NULL) {
+
+            /*Disable version should be part of the lib name
+            if (strncmp(pathToken, "version", 7) == 0) {
+                char *ver = strdup(pathToken);
+                char version[strlen(ver) - 9];
+                strncpy(version, ver+9, strlen(ver) - 10);
+                version[strlen(ver) - 10] = '\0';
+
+                strcat(lib, "-");
+                strcat(lib, version);
+            }*/
+            pathToken = strtok_r(NULL, ";", &path);
+        }
+
+        char *trimmedLib = utils_stringTrim(lib);
+        status = celix_module_loadLibraryForManifestEntry(module, trimmedLib, archive, &handle);
+
+        if ( (status == CELIX_SUCCESS) && (activator != NULL) && (strcmp(trimmedLib, activator) == 0) ) {
+            *activatorHandle = handle;
+        }
+        else if (handle!=NULL) {
+            celix_libloader_close(fwCtx, handle);
```

Comment:
`handle` has already been added to `module->libraryHandles`, closing it will lead to double-close in `celix_module_closeLibraries`. 

This is reported by @tira-misu in #479.